### ISSUE TITLE
Should abuse targeted at individual users be censored from CHQ's transcript?

### ISSUE THREAD

**Issue by Undo1** [Charcoal-SE/SmokeDetector#2192]
Occasionally, some idiot decides to exact 'revenge' (I guess?) on one of our members by posting the standard spread of nastiness directed at them.

SmokeDetector picks these up and faithfully posts them the CHQ. Should these posts be retroactively purged from the transcript?

**Comment by Videonauth** [comment]
If you speak of that chat transcript. Yes please! I think such stuff should nowhere linger around.

**Comment by ArtOfCode-** [comment]
What Magisch said. I'd advocate generally keeping it, but if the user wants it gone then we can delete or edit the message.

**Comment by Videonauth** [comment]
While I can understand the urge to keep it for the record. I can only talk about how I would feel myself if I would be the target of such harassment. I would at least be upset somehow and would not want it to be outside visible, because to get it for example of of google search engine it is a real pain and a lengthy process. So if we can avoid it to get indexed we should at least try to get rid of it before this happens.

**Comment by AWegnerGitHub** [comment]
It's not just Charcoal members that at targeted by trolls. We've picked up targeted attacks at members all over the network. 

We haven't removed these in the past. I don't think we should in the future. The only exception to this that we should make is if the target wishes for it to be removed. That's reasonable. 

---

That said, if we do vote to remove these, we need to answer some of the following so that we are consistent:

 - What does "targeted" mean? Does it mean a specific name? A specific group? A large group? Where is this line?
 - What type of content is removed? Do we remove just links to posts with bad content? Do we remove messages with the content displayed directly? 
 - How do we determine if one message is removed and another is not? Can you justify why one survives this process?
 - What is the time frame for removing a message? Do we remove it within 2 minutes? A day? Unlimited? 
 - Do we alert when a message is removed?

**Comment by surajrao** [comment]
Agreed with deleting from the transcript if the user requests. There is also the possibility of the content remaining in any other chat rooms that SD might have reported to.

**Comment by Undo1** [comment]
So, if the problem is Google indexing... could we do something clever with Cyrillic characters to hide from Google, with no further interruption to our day-to-day activities?

**Comment by ArtOfCode-** [comment]
I don't think the problem _is_ Google indexing. Google _does_ index SE chat, but you've got to be trying pretty hard with fairly specific search terms to get any results from it, IME. As far as I see it, the potential problems we have here are these:

 - **It's filth that we don't want to stick around longer than necessary.**  
   ...and we _do_ want spam to stick around longer than necessary? CHQ is _supposed_ to be a historical record of filth; if you're not comfortable seeing it, you're not going to last very long in CHQ. We shouldn't start deleting select types of filth and ignoring others; either we delete _all_ tp reports or we delete _none_.

 - **It's unpleasant for the victim to have to keep seeing it.**  
   This is a valid argument, but it may not apply to everyone. Personally, I couldn't care less if people throw abuse at me, and if someone starts targeting me with this stuff I won't have an issue with it sticking around in the transcript. However, not everyone has that attitude, so deleting it _if that's what the targeted user wants_ is a perfectly reasonable thing to do.

**TL;DR:** Don't delete it because it's abusive; delete it if the victim wants.

**Comment by ByteCommander** [comment]
Maybe we should not delete any posts but just redact user names or other identifying parts out of them in the transcript, on request by the targeted individual? 
Sure that would be more mod work than simply deleting the message, but it completely preserves the message edit history while hiding direct offense from the viewer. I don't think Google would index the old revisions either, but I don't know.

**Comment by iBug** [comment]
I agree that we should remove those from CHQ, but keeping it somewhere else like The Fire Department, for recording purposes, as we've been doing to normal reports in CHQ.
